src/EditorFeatures/Test2/IntelliSense/VisualBasicCompletionCommandHandlerTests.vb

```diff
@@ -472,16 +484,22 @@ End Class
                 ' allow the provider to continue
                 e.Set()
 
-                ' We should not have a session since we tear things down if we see a caret move
-                ' before the providers have returned.
-                Await state.AssertNoCompletionSession()
+                If completionImplementation = CompletionImplementation.Legacy Then
+                    ' We should not have a session since we tear things down if we see a caret move
+                    ' before the providers have returned.
+                    Await state.AssertNoCompletionSession()
+                Else
+                    ' Async provider can handle keys pressed while waiting for providers.
+                    Await state.AssertCompletionSession()
```

Comment:
this seems like we're breaking behavior.  The legacy test was intentional.  if the user navigates before completion is up, then we want that navigation to be respected.

Reply:
OK. Let us consider this once again at: [https://github.com/dotnet/roslyn/issues/31131](https://github.com/dotnet/roslyn/issues/31131)

---
In reply to: [232508407](https://github.com/dotnet/roslyn/pull/29016#discussion_r232508407) [](ancestors = 232508407)

Reply:
FWIW: this is a very important scenario.  I've included notes on this topic in the linked issue.  But this can seriously affect partner teams.  And i'm also concerned about how it will make things feel for C#/VB

Reply:
Thank you, @cyrusnajmabadi! I agree we should signoff this scenario with partner teams. I think we still can merge under a feature flag. 

---
In reply to: [232895699](https://github.com/dotnet/roslyn/pull/29016#discussion_r232895699) [](ancestors = 232895699)